### ISSUE TITLE
Impossible to unzip maps on android 10

### ISSUE THREAD

**Issue by Scafir** [junjunguo/PocketMaps#137]
Hey! 

I was messing around in Android Studio and realized that it is impossible to successfully retrieve a map on Android 10.0 (at least through the emulator). I digged around a little and got to this error message:
```
I/com.junjunguo.pocketmaps.downloader.MapDownloadUnzip: Unzipping map: africa_sao-tome-and-principe
W/System.err: java.io.FileNotFoundException: /storage/emulated/0/Download/pocketmaps/maps/africa_sao-tome-and-principe-gh/africa_sao-tome-and-principe.map: open failed: ENOENT (No such file or directory)
        at libcore.io.IoBridge.open(IoBridge.java:496)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:235)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:125)
        at com.junjunguo.pocketmaps.downloader.MapUnzip.extractFile(MapUnzip.java:84)
        at com.junjunguo.pocketmaps.downloader.MapUnzip.unzip(MapUnzip.java:52)
        at com.junjunguo.pocketmaps.downloader.MapDownloadUnzip$1.run(MapDownloadUnzip.java:65)
        at java.lang.Thread.run(Thread.java:919)
    Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
        at libcore.io.Linux.open(Native Method)
W/System.err:     at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at libcore.io.BlockGuardOs.open(BlockGuardOs.java:252)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:7255)
        at libcore.io.IoBridge.open(IoBridge.java:482)
    	... 6 more
I/com.junjunguo.pocketmaps.activities.DownloadMapActivity: Clearing dl file for map: africa_sao-tome-and-principe
```

It is worth noting that in order to display it, I had to add `e.printStackTrace()` at line 69 of MapDownloadUnzip.java.

Despite my efforts, I could not find why this happened, but it seems like on this particular version of Android, PocketMaps does not create the necessary folders before trying to write a file. This feeling is partially confirmed by a message appearing at the start of the app:
```
I/com.junjunguo.pocketmaps.activities.MainActivity: Warning: mapsFolder does not exist: /storage/emulated/0/Download/pocketmaps/maps
```

It is worth mentionning that this does not happen on API 28, both on hardware and in emulator.

Kind Regards

**Comment by Starcommander** [comment]
Hello @Scafir !

First of all: Thanks for opening this issue.
Is it possible to change the maps-directory on Android10 emu? (main menu)
Maybe unzipping is possible to a different directory. (write access)
And did you try to download different maps?
We can also try to modify the source to create the folder (and check if exists) before unzipping.
Also interesting: Logging whether the folder is writeable.

Best regards,
Paul

**Comment by Scafir** [comment]
Re!
Sorry for my late response, I had some things to sort out.

TLDR: since api 29, apps have [https://developer.android.com/about/versions/10/privacy/changes#scoped-storage](only) access to scoped storage. Since no other app uses PocketMaps' mapfiles, I would suggest removing the option of saving to the downloads entirely

So:
* Is it possible to change the maps-directory on Android10 emu? (main menu)
Yes, it is. If I select the app storage (both internal or sd card), it downloads and successfully unpacks it. I can open and navigate the map. 

* Maybe unzipping is possible to a different directory. (write access)
I am not too sure I did this correctly, but when I force the Documents directory (by changing line 35 of MapUnzip to `File dir = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS);`) to be the unzip folder, I still get the same error.

* And did you try to download different maps?
Yes I did. Always the same outcome

* We can also try to modify the source to create the folder (and check if exists) before unzipping.
I believe that this is already what the app is doing. If I manually create the folders, I still get the `I/com.junjunguo.pocketmaps.activities.MainActivity: Warning: mapsFolder does not exist: /storage/emulated/0/Download/pocketmaps/maps` when launching the app (and I still have the error)

* Logging whether the folder is writeable
This is the right call. We can see [https://developer.android.com/about/versions/10/privacy/changes#scoped-storage](here) that since api 29, any app has only scoped access to external storage! Indeed, Android Studio shows warnings on line 153 of MainActivity, saying that `Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS);` is deprecated.

Side note: I did not have to to look at it thoroughly, but killing the app while the map is still loading does some strange things. I will have "Map is not compatible with this version" and the Downloads folder selected as the destination folder when I launch it again. I then need to wait some time, kill the app, open it again to go back to nominal behaviour.

**Comment by Starcommander** [comment]
Hello @Scafir 

Thanks for your information, and testing.
I have created a new pre-release.
Maybe you can test the new apk:
https://github.com/junjunguo/PocketMaps/releases/tag/3.4-pre1
(Installing the apk may clear all settigs and maps of pocketmaps)

Thanks, and best regards,
Paul

**Comment by Starcommander** [comment]
Should be solved in V3.4
